src/Common/src/System/Net/DebugSafeHandle.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Win32.SafeHandles;
+using System.Diagnostics.CodeAnalysis;
+using System.Net.NetworkInformation;
+using System.Net.Sockets;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace System.Net
+{
+#if DEBUG
+    //
+    // This is a helper class for debugging GC-ed handles that we define.
+    // As a general rule normal code path should always destroy handles explicitly
+    //
+    internal abstract class DebugSafeHandle : SafeHandleZeroOrMinusOneIsInvalid
+    {
+        private string _trace;
+
+        protected DebugSafeHandle(bool ownsHandle) : base(ownsHandle)
+        {
+            Trace();
+        }
+
+        protected DebugSafeHandle(IntPtr invalidValue, bool ownsHandle) : base(ownsHandle)
+        {
+            SetHandle(invalidValue);
+            Trace();
+        }
+
+        private void Trace()
+        {
+            _trace = "WARNING! GC-ed  >>" + this.GetType().FullName + "<< (should be excplicitly closed) \r\n";
+#if TRAVE
```

Comment:
`TRAVE` (my guess: TRAce VErbose) is used throughout System.Net code. 

I found using these helpful during advanced async scenario debugging (e.g. Sockets IOCP implementation). 

The define enables extra tracing which might add overhead. The overheads are enough to ensure that nobody enable these by mistake (`DEBUG` and `TRACE` are too common).